src/analytics/navigationTracker/navigationTracker.test.ts

```diff
@@ -69,11 +69,26 @@ describe("navigationTracker", () => {
     href.dispatchEvent(action);
   });
 
-  //test trackNavigation accept link
-  test("trackNavigation should return false if a link without href", () => {
-    const href = document.createElement("A");
-    href.className = "govuk-footer__link";
-    href.innerHTML = "Link to GOV.UK";
+  //test trackNavigation accept navigation button
+  test("trackNavigation should return true if a navigation button", () => {
+    document.body.innerHTML = "<header></header><footer></footer>";
+    const href = document.createElement("BUTTON");
+    href.setAttribute("data-nav", "true");
+    href.setAttribute("data-link", "/next-url");
+    href.innerHTML = "Continue";
+    href.setAttribute("href", "http://localhost");
+    href.addEventListener("click", (event) => {
+      expect(newInstance.trackNavigation(event)).toBe(true);
+    });
+    href.dispatchEvent(action);
+  });
+
+  //test trackNavigation doesn't accept button
+  test("trackNavigation should return false if a navigation button", () => {
```

Comment:
Sorry, I am a bit confused. Is this test specifically checking whether the element is a button and, additionally, whether it does not have the (data-link) attribute, rather than just testing if it is a button?

Reply:
Sorry the test name was not good. "if it's not a navigation button" function needs to return false